src/Ocelot/DependencyInjection/ConfigurationBuilderExtensions.cs

```diff
@@ -50,9 +47,10 @@ public static IConfigurationBuilder AddOcelot(this IConfigurationBuilder builder
             var reg = new Regex(subConfigPattern, RegexOptions.IgnoreCase | RegexOptions.Singleline);
 
             var files = new DirectoryInfo(folder)
-                .EnumerateFiles()
-                .Where(fi => reg.IsMatch(fi.Name) && (fi.Name != excludeConfigName))
-                .ToArray();
+               .EnumerateFiles()
+               .Where(fi => reg.IsMatch(fi.Name) && (fi.Name != excludeConfigName)                               
+                    && (fi.Name.Contains(env.EnvironmentName) || fi.Name.Contains(globalConfigFile))) // Added to support sub services maping ex: ocelot.order.{EnvironmentName}.json

```

Comment:
Look at the line 45!
The old parameter `IWebHostEnvironment env` provides the same **EnvironmentName** property which you're trying to read from new `IHostEnvironment env`.

So, there is no need to change the type of `env` parameter at all!